File: windows/src/GoalMaker.Core/Connector/ConnectorLink.cs
namespace GoalMaker.Core.Connector;

/// <summary>A connector link as the owner may see it: when it was made, last used and revoked, never its secret.</summary>
public sealed record ConnectorLink(string Id, DateTimeOffset CreatedAt, DateTimeOffset? LastUsedAt, DateTimeOffset? RevokedAt)
{
    public bool Active => RevokedAt is null;
}

File: android/app/src/main/kotlin/com/goalmaker/app/application/connector/ConnectorLink.kt
package com.goalmaker.app.application.connector

import java.time.Instant

/** A connector link as the owner may see it: when it was made, last used and revoked, never its secret. */
data class ConnectorLink(
    val id: String,
    val createdAt: Instant,
    val lastUsedAt: Instant?,
    val revokedAt: Instant?,
) {
    val active: Boolean get() = revokedAt == null
}
